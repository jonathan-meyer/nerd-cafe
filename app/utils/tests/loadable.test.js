import React from 'react';
import { create } from 'react-test-renderer';
import loadable from 'utils/loadable';

describe('loadable', () => {
  it('should render and match the snapshot 1', async () => {
    const Foo = loadable(() => import('./Foo'));

    const root = await create(<Foo />);
    const fallback = root.toJSON();

    await Foo;
    const output = root.toJSON();

    expect(fallback).toMatchSnapshot();
    expect(output).toMatchSnapshot();
  });

  it('should render and match the snapshot 2', async () => {
    const Foo = loadable(() => import('./Foo'), { fallback: <div>Bar</div> });

    const root = await create(<Foo />);
    const fallback = root.toJSON();

    await Foo;
    const output = root.toJSON();

    expect(fallback).toMatchSnapshot();
    expect(output).toMatchSnapshot();
  });
});
