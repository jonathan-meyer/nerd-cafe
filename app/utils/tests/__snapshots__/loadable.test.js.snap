// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadable should render and match the snapshot 1 1`] = `null`;

exports[`loadable should render and match the snapshot 1 2`] = `
<div>
  Foo
</div>
`;

exports[`loadable should render and match the snapshot 2 1`] = `
<div>
  Bar
</div>
`;

exports[`loadable should render and match the snapshot 2 2`] = `
<div>
  Foo
</div>
`;
